jakarta/README.md

```diff
@@ -0,0 +1,103 @@
+# Jakarta migration
+
+This directory contains scripts and configuration files to automate the migration to Jakarta EE 9 (for now) and hopefully Jakarta EE 10 soon.
+
+## jakarta/transform.sh
+
+`jakarta/transform.sh` is the main script to run.
+It has to be run from the root of the Quarkus repository.
+
+It consists of several steps that leverage various tools:
+
+- [OpenRewrite](https://github.com/openrewrite/rewrite) - rewrites the POM files
+- [Eclipse Transformer](https://projects.eclipse.org/projects/technology.transformer) - rewrites the classes
+
+Note that this script also builds:
+
+- A patched version of OpenRewrite (while waiting for integration of our patches)
+- A patched version of the Rewrite Maven Plugin (to point to the patched OpenRewrite version)
+- A patched version of the Kotlin Maven Plugin (to allow skipping the `main` compilation, it's required for the OpenRewrite run)
+- SmallRye Config while waiting for a CR2
+
+## Approach
+
+### OpenRewrite
+
+The OpenRewrite transformation is done in one unique step to alleviate dependency issues we had before switching to this approach.
+You don't need to run OpenRewrite for the modules you add to `transform.sh`, you just need to adjust the configuration.
+
+OpenRewrite recipes are declared in the `jakarta/rewrite.yml` file.
+
+Documentation for writing recipes can be found:
+
+- In the [official documentation](https://docs.openrewrite.org/reference/recipes/maven)
+- In the [source for those not already released](https://github.com/gsmet/rewrite/tree/main/rewrite-maven/src/main/java/org/openrewrite/maven) (things are very simple there and well documented in the source)
+
+Recipes are not applied on all POM files, you need to declare the ones you want to apply to a given target POM when changes are necessary.
+Note that in most cases, you don't need to apply any changes.
+
+The active recipes have to be declared in the target POM with a `plugin` block in the `<build><plugins>` section similar to:
+
+```xml
```

Comment:
could we not avoid all the pom.xml changes by putting this plugin config in parent and have it pickup `rewrite.yml` ?

Reply:
ack - its just that if there is a rewrite.yml next to the pom.xml my jbang based runner for rewrite could be used in a parallel and removed need to change each pom.xml ...just add rewrite.yml where needed.